testing/scenario_app/android/app/src/main/java/dev/flutter/scenarios/MainActivity.java

```diff
@@ -1,12 +1,94 @@
 package dev.flutter.scenarios;
 
+import android.Manifest;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
 import android.os.Bundle;
+import android.os.Handler;
+import android.support.annotation.NonNull;
 
-import io.flutter.app.FlutterActivity;
+import java.io.FileDescriptor;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import io.flutter.Log;
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.android.FlutterFragment;
+import io.flutter.embedding.android.FlutterView;
+import io.flutter.embedding.engine.FlutterShellArgs;
+import io.flutter.embedding.engine.renderer.OnFirstFrameRenderedListener;
+import io.flutter.plugin.common.BasicMessageChannel;
+import io.flutter.plugin.common.BinaryCodec;
+
+public class MainActivity extends FlutterActivity implements OnFirstFrameRenderedListener {
+    final static String TAG = "Scenarios";
 
-public class MainActivity extends FlutterActivity {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        final Intent launchIntent = getIntent();
+        if ("com.google.intent.action.TEST_LOOP".equals(launchIntent.getAction())) {
+            if(Build.VERSION.SDK_INT > 22){
+                requestPermissions(new String[] {Manifest.permission.WRITE_EXTERNAL_STORAGE}, 1);
+            }
+            // Run for one minute, get the timeline data, write it, and finish.
+            final Uri logFileUri = launchIntent.getData();
+            new Handler().postDelayed(() -> writeTimelineData(logFileUri), 20000);
+        }
+    }
+
+    public void onFirstFrameRendered() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            reportFullyDrawn();
+        }
+    }
+
+    private FlutterShellArgs getFlutterShellArgs() {
+        FlutterShellArgs args = FlutterShellArgs.fromIntent(getIntent());
+        args.add(FlutterShellArgs.ARG_TRACE_STARTUP);
+        args.add(FlutterShellArgs.ARG_ENABLE_DART_PROFILING);
+        args.add(FlutterShellArgs.ARG_VERBOSE_LOGGING);
+
+        return args;
+    }
+
+    @Override
+    @NonNull
+    protected FlutterFragment createFlutterFragment() {
```

Comment:
Did you find that you really had to implement this method? It looks like everything you're passing to the FlutterFragment is the usual stuff...I think you should be able to remove this method entirely without impacting your solution. Please let me know if I'm missing anything.

Reply:
Yes, to override the shell args.

I filed a bug about that here: https://github.com/flutter/flutter/issues/36812

It's not the end of the world, but it might be nice if there was some way to more easily override the shell args.  Then again, I doubt most of our customers care about that.